misc/toml-parser/src/main/java/io/ballerina/toml/internal/parser/TomlParser.java

```diff
@@ -291,9 +296,9 @@ private STNode parseSingleKey() {
             case IDENTIFIER_LITERAL:
                 return parseIdentifierLiteral();
             case DOUBLE_QUOTE_TOKEN:
-            case TRIPLE_DOUBLE_QUOTE_TOKEN:
```

Comment:
Can we have a single key like this?
```toml
 [["""fruit"""]]
```
In this method we go to recovery with KEY_START ctx which allows both TRIPLE_DOUBLE_QUOTE_TOKEN and TRIPLE_SINGLE_QUOTE_TOKEN, but the parsing method does not allow them.

Therefore, for above example parseSingleKey will keep re-parsing again and again.

Reply:
It seems like we cant reuse same context for multi strings. Addressed this issue by introducing separate tokens for parsing multiline (literal) strings.